### PR TITLE
Ie 494

### PR DESCRIPTION
**Pull request by Vbhatia20**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55351)

### CODE REVIEW

File: 980-commons/src/main/java/io/harness/version/VersionInfoManagerV2.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+@Path("/version")
+@Produces(MediaType.APPLICATION_JSON)
+public class VersionInfoManagerV2 {
+  private static final String VERSION_YAML_FILE = "/opt/harness/version.yaml";
```
Comment (kapilgarg1996) [inline 980-commons/src/main/java/io/harness/version/VersionInfoManagerV2.java:27]: should this path come from a config ? If not, can we add this path in the ADR as well for future reference

File: 980-commons/src/main/java/io/harness/version/VersionInfoManagerV2.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+  @GET
+  public String getVersionInfo() {
+    try {
+      // Read the content of the version.yaml file
+      FileInputStream fileInputStream = new FileInputStream(VERSION_YAML_FILE);
+      String yamlContent = IOUtils.toString(fileInputStream, StandardCharsets.UTF_8);
+
+      // Parse the YAML content and create a JSON response
+      JSONObject json = new JSONObject();
```
Comment (kapilgarg1996) [inline 980-commons/src/main/java/io/harness/version/VersionInfoManagerV2.java:37]: We should use `snakeyaml` here. We already have it in our `WORKSPACE` so we can use that without adding more dependencies.

File: 980-commons/src/main/java/io/harness/version/VersionInfoManagerV2.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        if (parts.length == 2) {
+          String key = parts[0].trim();
+          String value = parts[1].trim();
+          json.put(key, value);
+        }
+      }
+
+      return json.toString();
+    } catch (IOException e) {
+      e.printStackTrace();
```
Comment (kapilgarg1996) [inline 980-commons/src/main/java/io/harness/version/VersionInfoManagerV2.java:49]: lets use logger here to log the stacktrace in appropriate format.

File: 270-verification/chart/values.yaml [hunk 4]
```diff
@@ -33,6 +33,24 @@ global:
       userKey: ""
       passwordKey: ""
       extraArgs: ""
+      secrets:
```
Comment (kapilgarg1996) [inline 270-verification/chart/values.yaml:36]: this change seems unrelated, shall we move it to a separate PR ?

File: 980-commons/src/main/java/io/harness/version/VersionInfoManagerV2.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+import org.json.JSONObject;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+@Path("/version")
+@Produces(MediaType.APPLICATION_JSON)
+public class VersionInfoManagerV2 {
```
Comment (kapilgarg1996) [inline 980-commons/src/main/java/io/harness/version/VersionInfoManagerV2.java:26]: Shouldn't we rename it to `VersionInfoResourceV2` ?

File: access-control/service/src/main/java/io/harness/accesscontrol/commons/version/VersionInfoResourceV2.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+import io.harness.version.VersionInfoManagerV2;
+import io.harness.version.VersionInfoV2;
+
+import com.google.inject.Inject;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.*;
```
Comment (xingchi-jin) [inline access-control/service/src/main/java/io/harness/accesscontrol/commons/version/VersionInfoResourceV2.java:20]: avoid using star
